erizo/src/erizo/WebRtcConnection.cpp

```diff
@@ -222,46 +239,82 @@ bool WebRtcConnection::werePacketLossesRecently() {
-    connection->media_streams_.push_back(media_stream);
+    ELOG_DEBUG("%s message: Adding mediaStream, id: %s, transceivers_count: %d",
+      connection->toLog(), media_stream->getId().c_str(), connection->transceivers_.size());
+    bool added = false;
+    std::for_each(connection->transceivers_.begin(), connection->transceivers_.end(),
+        [&media_stream, &added, connection] (const auto &transceiver) {
+      if (!transceiver->hasMediaStream() && !added) {
+        transceiver->setMediaStream(media_stream);
+        added = true;
+        ELOG_DEBUG("%s message: Added mediaStream reusing transceiver, id: %s, transceiverId: $d",
+          connection->toLog(), media_stream->getId().c_str(), transceiver->getId());
+      }
+    });
+    if (!added) {
+      auto transceiver = std::make_shared<Transceiver>(std::to_string(connection->transceivers_.size()));
+      transceiver->setMediaStream(media_stream);
+      connection->transceivers_.push_back(transceiver);
+      ELOG_DEBUG("%s message: Added mediaStream, id: %s, transceiverId: %d",
+        connection->toLog(), media_stream->getId().c_str(), transceiver->getId());
+    }
   });
 }
 
 boost::future<void> WebRtcConnection::removeMediaStream(const std::string& stream_id) {
   return asyncTask([stream_id] (std::shared_ptr<WebRtcConnection> connection) {
     boost::mutex::scoped_lock lock(connection->update_state_mutex_);
     ELOG_DEBUG("%s message: removing mediaStream, id: %s", connection->toLog(), stream_id.c_str());
-    connection->media_streams_.erase(std::remove_if(connection->media_streams_.begin(),
-                                                    connection->media_streams_.end(),
-      [stream_id, connection](const std::shared_ptr<MediaStream> &stream) {
-        bool isStream = stream->getId() == stream_id;
-        if (isStream) {
-          auto video_it = connection->local_sdp_->video_ssrc_map.find(stream->getLabel());
-          if (video_it != connection->local_sdp_->video_ssrc_map.end()) {
-            connection->local_sdp_->video_ssrc_map.erase(video_it);
-          }
-          auto audio_it = connection->local_sdp_->audio_ssrc_map.find(stream->getLabel());
-          if (audio_it != connection->local_sdp_->audio_ssrc_map.end()) {
-            connection->local_sdp_->audio_ssrc_map.erase(audio_it);
-          }
+    auto transceiver_it = std::find_if(connection->transceivers_.begin(), connection->transceivers_.end(),
+      [stream_id] (const std::shared_ptr<Transceiver> &transceiver) {
+        if (transceiver->hasMediaStream()) {
+          return transceiver->getMediaStream()->getId() == stream_id;
         }
-        return isStream;
-      }));
+        return false;
     });
+    if (transceiver_it != connection->transceivers_.end()) {
+      std::shared_ptr<MediaStream> stream = (*transceiver_it)->getMediaStream();
+      (*transceiver_it)->resetMediaStream();
+      bool isStream = stream->getId() == stream_id;
```

Comment:
Is this a sanity check? Wouldn't we know by now that this stream is the stream we want to remove? Just checking I'm not missing something

Reply:
yes, you're right, this is totally unncessary